src/compile/layoutsize/parse.ts

```diff
@@ -19,16 +19,12 @@ export function parseLayerLayoutSize(model: Model) {
 
 export const parseRepeatLayoutSize = parseLayerLayoutSize;
 
-const SIZE_TYPE_TO_MERGE = {
-  vconcat: 'width',
-  hconcat: 'height'
-};
-
 export function parseConcatLayoutSize(model: ConcatModel) {
   parseChildrenLayoutSize(model);
   const layoutSizeCmpt = model.component.layoutSize;
 
-  const sizeTypeToMerge = SIZE_TYPE_TO_MERGE[model.concatType];
+  const columns = model.layout.columns;
+  const sizeTypeToMerge = columns === 1 ? 'width' : columns === undefined ? 'height' : undefined;
```

Comment:
change `concatType` to a getter (which determines type based on column)? 

So here we can make the code a bit more readable?

Reply:
Would that actually be more readable? I feel this code is very clear about what it does and you don't need to look at another function. I can make this a case statement, though.

Reply:
still irrelevant to this PR, but sure I can ignore it.